lib/manager/gomod/artifacts.ts

```diff
@@ -66,6 +66,8 @@ export async function updateArtifacts({
       },
       docker: {
         image: 'renovate/go',
+        tagConstraint: config.compatibility?.go,
+        tagScheme: 'npm',
```

Comment:
`npm` ?

Reply:
Otherwise `1.14` is considered an invalid version if using regular `semver`. `npm` allows it and matches it as `1.14.*`